nesis/api/core/document_loaders/sharepoint.py

```diff
@@ -23,158 +27,278 @@
 _sharepoint_context = None
 
 
-def fetch_documents(**kwargs) -> None:
+def fetch_documents(
+    connection: Dict[str, str],
+    rag_endpoint: str,
+    http_client: http.HttpClient,
+    metadata: Dict[str, Any],
+    cache_client: memcache.Client,
+) -> None:
+    global _sharepoint_context
     try:
-        rag_endpoint = kwargs["rag_endpoint"]
-        connection = kwargs["connection"]
+
+        site_url = connection.get("site_url")
+        client_id = connection.get("client_id")
+        tenant = connection.get("tenant")
+        thumbprint = connection.get("thumbprint")
+        cert_path = connection.get("certificate_path")
```

Comment:
@akizito for this, I think that we should have the certificate contents instead of the path. To use them, we should have to persist them into a temporary file and pass the path to the file into the `cert_path` parameter. This way, the connection details can be stored and retrieved from the database.